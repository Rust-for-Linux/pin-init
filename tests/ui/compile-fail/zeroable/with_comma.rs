#![feature(allocator_api)]
extern crate pinned_init;
use pinned_init::*;

#[derive(Zeroable)]
struct Foo {
    a: usize,
    b: usize,
}

fn main() {
    let _ = init!(Foo {
        a: 0,
        ..Zeroable::zeroed(),
    });
}
