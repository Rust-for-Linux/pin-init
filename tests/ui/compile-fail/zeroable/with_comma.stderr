error: no rules expected `,`
  --> tests/ui/compile-fail/zeroable/with_comma.rs:11:13
   |
11 |       let _ = init!(Foo {
   |  _____________^
12 | |         a: 0,
13 | |         ..Zeroable::zeroed(),
14 | |     });
   | |______^ no rules expected this token in macro call
   |
note: while trying to match `)`
  --> src/macros.rs
   |
   |         @munch_fields($(..Zeroable::zeroed())? $(,)?),
   |                                                     ^
   = note: this error originates in the macro `$crate::__init_internal` which comes from the expansion of the macro `init` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected `,`
  --> tests/ui/compile-fail/zeroable/with_comma.rs:11:13
   |
11 |       let _ = init!(Foo {
   |  _____________^
12 | |         a: 0,
13 | |         ..Zeroable::zeroed(),
14 | |     });
   | |______^ no rules expected this token in macro call
   |
note: while trying to match `)`
  --> src/macros.rs
   |
   |         @munch_fields(..Zeroable::zeroed() $(,)?),
   |                                                 ^
   = note: this error originates in the macro `$crate::__init_internal` which comes from the expansion of the macro `init` (in Nightly builds, run with -Z macro-backtrace for more info)
